compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -463,6 +466,63 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
+        let hir = self.infcx.tcx.hir();
+        let mir_hir_id = self.mir_hir_id();
+        let node = hir.item(mir_hir_id);
+        if let hir::ItemKind::Fn(_, _, body_id) = node.kind {
+            let body = hir.body(body_id);
+            if let hir::ExprKind::Block(block, _) = body.value.kind {
+                if let Some(expr) = block.expr {
+                    if let hir::ExprKind::DropTemps(expr) = expr.kind {
+                        if let hir::ExprKind::Match(expr, ..) = expr.kind {
+                            if let hir::ExprKind::Call(_, [expr, ..]) = expr.kind {
+                                if let hir::ExprKind::MethodCall(path_segment, ..) = expr.kind {
+                                    if let Some(hir_id) = path_segment.hir_id {
+                                        let typeck = self.infcx.tcx.typeck(hir_id.owner);
+                                        if let Some(def_id) =
+                                            typeck.type_dependent_def_id(expr.hir_id)
+                                        {
+                                            if let Some(def_id) =
+                                                self.infcx.tcx.impl_of_method(def_id)
+                                            {
+                                                let assoc_items =
+                                                    self.infcx.tcx.associated_items(def_id);
+                                                let suggestions = assoc_items
+                                                    .in_definition_order()
+                                                    .filter_map(|assoc_item_def| {
+                                                        let original_method_ident =
+                                                            path_segment.ident;
+                                                        let ident = assoc_item_def.ident;
+                                                        if original_method_ident != ident
+                                                            && ident.as_str().starts_with(
+                                                                &original_method_ident
+                                                                    .name
+                                                                    .to_string(),
+                                                            )
+                                                        {
+                                                            return Some(format!("{}()", ident));
+                                                        }
+                                                        None
+                                                    });
+                                                err.span_suggestions(
+                                                    path_segment.ident.span,
+                                                    &format!("use mutable method"),
```

Comment:
I'm surprised that we don't check that the methods we're suggesting are actually yielding mutable references. Is that something which is just too tricky to check here?

Reply:
This method called only [here](https://github.com/rust-lang/rust/pull/81466/files#diff-d312b694654d74c5ee17332e226377520e7a0b67f29f823ffb68df6ca883844aR344).
It seems we don't need to check.

Reply:
Could you add a comment that documents this assumption?